src/issuance.rs

```diff
@@ -308,11 +337,9 @@ impl IssueBundle<Prepared> {
         let expected_ik: IssuanceValidatingKey = (isk).into();
 
         // Make sure the `expected_ik` matches the `asset` for all notes.
-        self.actions.iter().try_for_each(|action| {
-            action
-                .are_note_asset_ids_derived_correctly(&expected_ik)
-                .map(|_| ()) // Transform Result<NoteType,Error> into Result<(),Error)>.
-        })?;
+        self.actions
+            .iter()
+            .try_for_each(|action| action.verify_supply(&expected_ik).map(|_| ()))?;
```

Comment:
I'm open to suggestions on how to avoid `.map(|_| ())`

Reply:
By the way, do we need to perform all verification checks here or check for uniqueness of asset base only?